AI/VCAI/Pathfinding/AINodeStorage.cpp

```diff
@@ -342,19 +342,11 @@ bool AINodeStorage::hasBetterChain(const PathNodeInfo & source, CDestinationNode
 	return false;
 }
 
-bool AINodeStorage::isTileAccessible(int3 pos, const EPathfindingLayer layer) const
+bool AINodeStorage::isTileAccessible(const int3 & pos, const EPathfindingLayer layer) const
 {
-	auto chains = nodes[pos.x][pos.y][pos.z][layer];
-
-	for(const AIPathNode & node : chains)
-	{
-		if(node.action != CGPathNode::ENodeAction::UNKNOWN)
-		{
-			return true;
-		}
-	}
+	auto node = nodes[pos.x][pos.y][pos.z][layer][0];
```

Comment:
Is this `auto` copy of reference?

Reply:
Should I use auto & here? And should I use the same structure in other places where I use not all 5 dimensions? For instance here? It will be boost::multi_array::sub_array
auto chains = nodes[pos.x][pos.y][pos.z][layer];

Reply:
Done